### PR TITLE
Mass edit migration

### PR DESCRIPTION
**Pull request by flotho**

### CODE REVIEW

File: mass_editing/__openerp__.py [hunk 1]
```diff
@@ -21,10 +21,11 @@
 ##############################################################################
 {
```
Comment (elicoidal) [inline mass_editing/__openerp__.py:22]: Thanks for your contribution!

Please use simple headers following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: mass_editing/__openerp__.py [hunk 2]
```diff
@@ -21,10 +21,11 @@
...
     "name": "Mass Editing",
-    "version": "8.0.1.3.0",
+    "version": "9.0.1.1.0",
     "author": "Serpent Consulting Services,Odoo Community Association (OCA)",
     "contributors": [
         "Oihane Crucelaegui <oihanecrucelaegi@gmail.com>",
+        "Florent Thomas <contact@mind-and-go.com>"
```
Comment (elicoidal) [inline mass_editing/__openerp__.py:28]: Please create a proper README file following https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md and https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
Remove description key

File: mass_editing/__openerp__.py [hunk 3]
```diff
@@ -21,10 +21,11 @@
 ##############################################################################
 {
     "name": "Mass Editing",
-    "version": "8.0.1.3.0",
+    "version": "9.0.1.1.0",
```
Comment (hbrunn) [inline mass_editing/__openerp__.py:24]: why not 9.0.1.0.0?

File: mass_editing/models/ir_model_fields.py [hunk 4]
```diff
@@ -20,15 +20,16 @@
...
+_logger = logging.getLogger(__name__)
 
-class IrModelFields(orm.Model):
+class IrModelFields(models.Model):
     _inherit = 'ir.model.fields'
 
-    def search(
-            self, cr, uid, args, offset=0, limit=0, order=None, context=None,
-            count=False):
+    @api.model
```
Comment (hbrunn) [inline mass_editing/models/ir_model_fields.py:31]: add `@api.returns`

File: mass_editing/models/ir_model_fields.py [hunk 5]
```diff
@@ -20,15 +20,16 @@
 #
 ##############################################################################
 
-from openerp.osv import orm
+import logging
+from openerp import models, fields, api, _
```
Comment (hbrunn) [inline mass_editing/models/ir_model_fields.py:24]: most of those imports are unused

File: mass_editing/models/mass_object.py [hunk 6]
```diff
@@ -20,60 +20,66 @@
...
-        'model_id': fields.many2one(
-            'ir.model', 'Model', required=True, select=1),
-        'field_ids': fields.many2many(
-            'ir.model.fields', 'mass_field_rel', 'mass_id', 'field_id',
-            'Fields'),
-        'ref_ir_act_window': fields.many2one(
-            'ir.actions.act_window', 'Sidebar Action', readonly=True,
+
+
+    name = fields.Char("Name", size=64, required=True, select=1)
```
Comment (hbrunn) [inline mass_editing/models/mass_object.py:35]: size is not mandatory any more, so remove it where not functionally sensible

File: mass_editing/models/mass_object.py [hunk 7]
```diff
@@ -20,60 +20,66 @@
...
-        if not model_id:
-            return {'value': {'model_ids': [(6, 0, [])]}}
-        model_ids = [model_id]
-        model_obj = self.pool['ir.model']
-        active_model_obj = self.pool.get(model_obj.browse(
-            cr, uid, model_id).model)
+    @api.onchange('model_id')
+    def onchange_model_id(self):
+        if not self.model_id:
+            self.model_ids = [(6, 0, [])]
```
Comment (hbrunn) [inline mass_editing/models/mass_object.py:58]: you don't need this line

File: mass_editing/models/mass_object.py [hunk 8]
```diff
@@ -20,60 +20,66 @@
...
+            return self
+        model_ids = [self.model_id.id]
+        model_obj = self.env['ir.model']
+        _logger.debug("MODEL")
+        _logger.debug(model_obj.browse(self.model_id.id).model)
+        active_model_obj = self.env[model_obj.browse(self.model_id.id).model]
```
Comment (hbrunn) [inline mass_editing/models/mass_object.py:64]: `self.model_id` already is your browse record, no need to rebrowse

File: mass_editing/models/mass_object.py [hunk 9]
```diff
@@ -20,60 +20,66 @@
...
         if active_model_obj._inherits:
             for key, val in active_model_obj._inherits.items():
-                found_model_ids = model_obj.search(
-                    cr, uid, [('model', '=', key)], context=context)
-                model_ids += found_model_ids
-        return {'value': {'model_ids': [(6, 0, model_ids)]}}
+                found_model_ids = model_obj.search([('model', '=', key)])
+                _logger.debug("found_model_ids" )
+                _logger.debug(found_model_ids)
+                model_ids += [m.id for m in found_model_ids]
```
Comment (hbrunn) [inline mass_editing/models/mass_object.py:71]: `found_model_ids.ids`

File: mass_editing/models/mass_object.py [hunk 10]
```diff
@@ -20,60 +20,66 @@
...
+        self.model_ids = [(6, 0, model_ids)]
```
Comment (hbrunn) [inline mass_editing/models/mass_object.py:72]: you can simply assign a browse record

File: mass_editing/models/mass_object.py [hunk 11]
```diff
@@ -96,43 +100,41 @@ def create_action(self, cr, uid, ids, context=None):
...
-            context)
+            })
         return True
 
-    def unlink_action(self, cr, uid, ids, context=None):
-        for template in self.browse(cr, uid, ids, context=context):
+    @api.multi
+    def unlink_action(self):
+        for template in self:
+            template = self
```
Comment (hbrunn) [inline mass_editing/models/mass_object.py:116]: this line assigns the original browse record again, which fails below for len(self) > 1. remove

File: mass_editing/models/mass_object.py [hunk 12]
```diff
@@ -96,43 +100,41 @@ def create_action(self, cr, uid, ids, context=None):
...
+    def unlink(self):
+        self.unlink_action()
+        return super(MassObject, self).unlink()
 
-    def unlink(self, cr, uid, ids, context=None):
-        self.unlink_action(cr, uid, ids, context=context)
-        return super(MassObject, self).unlink(cr, uid, ids, context=context)
-
-    def copy(self, cr, uid, record_id, default=None, context=None):
+    @api.one
```
Comment (hbrunn) [inline mass_editing/models/mass_object.py:135]: add `@api.returns`

File: mass_editing/models/mass_object.py [hunk 13]
```diff
@@ -96,43 +100,41 @@ def create_action(self, cr, uid, ids, context=None):
...
+    def copy(self, default=None):
         if default is None:
             default = {}
         default.update({'name': '', 'field_ids': []})
-        return super(MassObject, self).copy(
-            cr, uid, record_id, default, context=context)
+        return super(MassObject, self).copy(record_id, default)
```
Comment (hbrunn) [inline mass_editing/models/mass_object.py:140]: remove `record_id`

**Comment by hbrunn** [comment]
you should also remove most of the debug logging and fix what travis complains about

**Comment by flotho** [comment]
@JayVora-SerpentCS  is mostly most efficient to port his own module thanks to #315 .
So I close this one

**Pull request closed without merging**